lib/cisco_node_utils/tacacs_server_host.rb

```diff
@@ -23,41 +23,107 @@ class TacacsServerHost < NodeUtil
     attr_reader :name
     @hosts = {}
 
-    def initialize(name, create=true)
+    def initialize(name, instantiate=true, host_port=nil)
       fail TypeError unless name.is_a? String
       fail ArgumentError if name.empty?
       @name = name
-      return unless create
-      # 'feature tacacs+' must be enabled to create a host
-      TacacsServer.new.enable unless TacacsServer.enabled
-      config_set('tacacs_server_host', 'host', '', name)
+
+      if platform == :ios_xr
+        if host_port.nil?
+          @port = config_get_default('tacacs_server_host', 'port')
+        else
+          fail ArgumentError, 'host_port must be an Integer' \
+            unless host_port.is_a?(Integer)
+          @port = host_port
+        end
+      end
+
+      create if instantiate
+
+      return if platform == :ios_xr
+
+      if instantiate
+        # 'feature tacacs+' must be enabled to create a host
+        TacacsServer.new.enable unless TacacsServer.enabled
+      end
+
+      return if host_port.nil?
+      fail ArgumentError, 'host_port must be an Integer' \
+        unless host_port.is_a?(Integer)
+      self.port = host_port
     end
 
     def self.hosts
-      @hosts = {}
+      hosts = {}
+
+      hosts_list = config_get('tacacs_server_host', 'hosts')
+      return hosts if hosts_list.empty?
 
-      return @hosts unless TacacsServer.enabled
+      hosts_list.each do |name|
+        if platform == :ios_xr
+          host_port = config_get('tacacs_server_host', 'port', ip: name)
+          host_port = host_port[0] if host_port.is_a?(Array)
+          host_port = host_port.to_i
 
-      hosts = config_get('tacacs_server_host', 'hosts')
-      unless hosts.nil?
-        hosts = [hosts] if hosts.is_a?(Hash)
-        hosts.each do |name|
-          @hosts[name] = TacacsServerHost.new(name, false) if @hosts[name].nil?
+          hosts[name] = TacacsServerHost.new(name, false, host_port)
+        else
+          hosts[name] = TacacsServerHost.new(name, false) if @hosts[name].nil?
         end
       end
-      @hosts
+      hosts
+    end
+
+    def create
+      destroy if platform == :ios_xr
+      config_set('tacacs_server_host',
+                 'host',
+                 state: '',
+                 ip:    name,
+                 port:  @port)
```

Comment:
I think you need a separate setter for NX and XR. I'm fairly certain this will fail on NX since you're supplying an extra key:

 ```
host:
  nexus:
    set_value: "<state> tacacs-server host <ip>"
```